### PR TITLE
Add support for Redis replication

### PR DESCRIPTION
**Pull request by krallin**
Add support for Redis replication. This again follows the same pattern set forward in https://github.com/aptible/docker-postgresql/pull/19.

This one is quite trivial (Redis replication requires little setup).

This time my test script is:

```
set -o errexit
set -o nounset
set -o xtrace

space () {
  echo
  echo
  echo
  echo
  echo
  echo
  echo
  echo
}

IMG="quay.io/aptible/redis:2.8"

DATA="/tmp/docker-redis/data"
MASTER_CONTAINER="redis-master"
SLAVE_CONTAINER="redis-slave"

# Cleanup
docker rm -f "$MASTER_CONTAINER" "$SLAVE_CONTAINER" || true
docker-machine ssh docker-vm sudo rm -rf "${DATA}"

make build

PASSPHRASE=testpass

space
echo "Initializing master"
docker run -it --rm -e PASSPHRASE="${PASSPHRASE}" -v "${DATA}/master:/var/db" "$IMG" --initialize

space
echo "Starting master"
MASTER_PORT=63791
docker run -d --name="${MASTER_CONTAINER}" \
  -e "PORT=$MASTER_PORT" \
  -v "${DATA}/master:/var/db" \
  "${IMG}"

MASTER_IP="$(docker inspect --format '{{ .NetworkSettings.IPAddress }}' "$MASTER_CONTAINER")"
MASTER_URL="redis://:$PASSPHRASE@$MASTER_IP:$MASTER_PORT"

# Add data before starting replication
docker run -it --rm "$IMG" --client "$MASTER_URL" SET test_before TEST_DATA

# Start replication
space
echo "Activating leader"
eval "$(docker run -it --rm "$IMG" --activate-leader "$MASTER_URL" | dos2unix)"


space
echo "Initializing slave"
echo "Master Url: $REDIS_REPLICATION_MASTER_URL"

docker run -it --rm \
  -e "REDIS_REPLICATION_MASTER_URL=$REDIS_REPLICATION_MASTER_URL" \
  -v "${DATA}/slave:/var/db" \
  "$IMG" --initialize-follower


space
echo "Starting slave"
SLAVE_PORT=63792
docker run -d --name "$SLAVE_CONTAINER" \
  -e "PORT=$SLAVE_PORT" \
  -v "${DATA}/slave:/var/db" \
  "$IMG"


SLAVE_IP="$(docker inspect --format '{{ .NetworkSettings.IPAddress }}' "$SLAVE_CONTAINER")"
SLAVE_URL="redis://:$PASSPHRASE@$SLAVE_IP:$SLAVE_PORT"

# Add data now that replication has started
docker run -it --rm "$IMG" --client "$MASTER_URL" SET test_after TEST_DATA

# Check the data now

docker run -it --rm "$IMG" --client "$MASTER_URL" GET test_before | grep "TEST_DATA"
docker run -it --rm "$IMG" --client "$MASTER_URL" GET test_after  | grep "TEST_DATA"

echo "Test OK!"
```

Same question as in https://github.com/aptible/docker-mysql/pull/15, though: do we *want* to run tests in-container? Running them outside of the container might be simpler *and* more realistic.

Cheers,

### CODE REVIEW

**Comment by krallin** [comment]
Heh, the tests I added in #11 helped me find a (nasty) bug here. I've rebased this branch of top of #11.

Cheers,

**Comment by krallin** [comment]
@fancyremarker @aaw,

This should be ready for review! 

The replication API is an `--initialize-from` `run-database.sh` command, which replaces `--initialize` (as the name suggests), and expects a database URL to the master (we might also want to expose environment variables provided to the master when it initialized, though so far I haven't needed that for any DB).

---

There will be a few Travis-related changes to make before merging:

- Update Docker credentials to use Aptible credentials (currently I'm using a robot assigned to my account on Quay and my account on the Docker hub).
- Update branch being pushed.

Cheers,

File: bin/run-database.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+set -o errexit
+
+
+  # shellcheck disable=SC1091
```
Comment (fancyremarker) [inline bin/run-database.sh]: Is shellcheck being run automatically?

**Comment by fancyremarker** [comment]
This one looks excellent too!

**Comment by krallin** [comment]
So, this is turning into a bit of a mega-PR, but I've added thrown in a fix for #8 in here as well so I can properly test that my `--initialize-from` changes in Sweetness aren't breaking anything in clone.

**Comment by krallin** [comment]
@fancyremarker Noticed you asked whether shellcheck was being run automatically: no, not yet, but I'd like to try and do so.

Cheers,

**Pull request merged**